File: python/two_pointer/valid_palindrome.py

import re;

def is_valid(s: str) -> bool:

	txt = s.lower()
	left = 0 
	right = len(s) - 1

	while left < right:

		while left < right and not re.findall('/a-z0-9/',txt[left]):
			left+=1

		while left < right and not re.findall('/a-z0-9/',txt[right]):
			right -=1

		if txt[left] != txt[right] :
			return False

		left += 1
		right -= 1
	return True	

File: js/src/TwoPointer/ValidPalindrome.js



export const isvalid = (s)=>{

	const txt = s.toLowerCase(s);
	let left = 0;
	let right = txt.length -1;

	while(left < right){

		while(left < right && !/a-za0-9/.test(txt[left])){
		
			left++;
		
		}

		while(left < right && !/a-z0-9/.test(txt[right])){
			
			right--;
		
		}

		if(txt[left] !== txt[right]){
			return false;
		}

		left++;
		right--;
	}
	return true;
}